cmake/DAGMC_macros.cmake

```diff
@@ -206,7 +207,10 @@ macro (dagmc_install_library lib_name)
                    INSTALL_RPATH_USE_LINK_PATH TRUE)
     endif ()
     target_link_libraries(${lib_name}-shared ${LINK_LIBS_SHARED})
+    target_include_directories(${lib_name}-shared INTERFACE ${CMAKE_INSTALL_PREFIX}/${INSTALL_INCLUDE_DIR}
+                                                            ${MOAB_INCLUDE_DIRS})
```

Comment:
What is the purpose of this line? This seems to be including the directory where the headers will be installed after compiling is finished. This is what is causing all those build warnings.

Reply:
This call ties the location of the installed headers to the library target `lib_name` so that when targets are exported the external targets linking to them can find those headers without relying on a variable like `DAGMC_INCLUDE_DIRS`. When these lines are commented, external targets will fail immediately b/c they can't find the necessary header files.

BUT I've updated those calls so that during a build the headers are found in the build location with an updated location for the installation location when the target is exported.